### DISCUSSION TITLE
ifnot

### DISCUSSION THREAD

**Discussion by kskog** [dotnet/csharplang#7817] [Language Ideas] [answered]
Usually, I try to reduce nesting by checking conditions and returning as quickly as possible. This requires inverting of if statements. To improve readability, maybe you could add an ifnot or if! statement.

instead of 

```
if (!(ev.Device.Type == DeviceType.Gateway && ev.PropertyId == MtIdentifier.NetworkMode && ev.IsChanged))
{
  return;
}

// ...
```

one could write like this:

```
ifnot (ev.Device.Type == DeviceType.Gateway && ev.PropertyId == MtIdentifier.NetworkMode && ev.IsChanged)
{
  return;
}

// ...
```

**Comment by ufcpp** [comment, marked as answer]
see https://github.com/dotnet/csharplang/issues/882

**Comment by jnm2** [comment]
This is a side topic, but I find this an enjoyable way to handle examples like the one you shared. It's not always applicable, though.

```cs
if (ev is not { Device.Type: DeviceType.Gateway, PropertyId: MtIdentifier.NetworkNode, IsChanged: true })
```
